sml_builder/cms.py

```diff
@@ -0,0 +1,44 @@
+import os
+
+import contentful
+
+# Will need a official contentful account for the project/ team, and then we can update
+# the space_id, content_delivery_api_key
+SPACE_ID = os.environ.get("SPACE_ID")
+CDA_KEY = os.environ.get("CDA_KEY")
+client = contentful.Client(SPACE_ID, CDA_KEY)
+
+
+# Returns the content depending on the content type
+def getContent(contentType):
+    content_type = client.content_type(contentType)
+    # print("content_type: ", content_type)
+    entries_by_content_type = getEntriesByContentType(content_type)
+    # print("entries_by_content_type: ", entries_by_content_type)
+
+    content = compileContent(entries_by_content_type)
+
+    return content
+
+
+# Gets the entries for the type of content passed into the function
+def getEntriesByContentType(contentType):
+    entries_by_content_type = client.entries({"content_type": contentType.id})
+    # print("entries_by_content_type2: ", entries_by_content_type)
+
+    return entries_by_content_type
+
+
+# Returns all the content depending if it's a list of content such as items on a table
+# or single item
+def compileContent(entries_by_content_type):
+    entry_fields = []
+    if (len(entries_by_content_type)) == 1:
+        entry_fields = entries_by_content_type[0].fields()
+    else:
+        for entry in entries_by_content_type:
```

Comment:
> What does the structure look like here? I am wondering why the for loop doesn't just work regardless of if we have a single entry or multiple?

Hey Steve, this was me trying to test something or make something work not sure this was written a long time ago but you're right we can refactor this.

This can be done with the code below:
 ```
entry_fields = []
    for entry in entries_by_content_type:
        entry_fields.append(entry.fields())
  return entry_fields
```


Once the change above is made, we need to make some minor changes to pages where we are only retrieving one entry such as the index.html page/ home page, where instead of `hero_title` you would need to replace it with `content[0]["hero_title"]` as now regardless of how many entries are being retrieved, it will be put in a list.

Hope this helps.

Reply:
This is for convenient referencing later on, for example if we had a list of table items and we wanted to reference a field in one of the table items we would do entry_by_content_type.entry.entry_field. For a content type with only one entry we would still have entry_by_content_type.entry.entry_field, however since the entry_by_content_type dictionary only contains one entry it makes more sense to return without the dictionary wrapper as entry.entry_field

Reply:
I might be misunderstanding your response, but what I meant was if you just do:

for entry in entries_by_content_type:
   entry_fields.append(entry.fields())
   
Wouldn't this work even in the scenario where there was only one element in the entries_by_content_type ?

Reply:
Yes that would work, for a single entry it would return in the format [entry] as opposed to entry. However, where these single entries are returned, the code is expecting the entry object itself rather than a list. This means we would have to use entry_list[0] everywhere where we are expecting a single entry after calling the function. To save a line of code every time we call this function it makes more sense to just remove the list wrapper within the function itself.